global/scripts/MyelinMap_BC.sh

```diff
@@ -0,0 +1,180 @@
+
+#mandatory
+#general inputs
+opts_AddMandatory '--study-folder' 'StudyFolder' 'path' "folder that contains all subjects"
+opts_AddMandatory '--subject' 'Subject' '100206' "one subject ID"
+opts_AddMandatory '--registration-name' 'RegName' 'MSMAll' "the registration string corresponding to the input files, e.g. 'MSMAll' or ''"
+opts_AddMandatory '--input-mesh' 'InputMesh' 'meshnum or native' "input resolution mesh node count (in thousands) or 'native', like '164' for 164k_fs_LR"
+opts_AddMandatory '--output-mesh' 'OutputMesh' 'meshnum or native' "low resolution mesh node count (in thousands) or 'native', like '32' for 32k_fs_LR"
+#optional inputs
+opts_AddOptional '--use-ind-mean' 'UseIndMean' 'YES or NO' "whether to use the mean of the subject's myelin map as reference map's myelin map mean , defaults to 'NO'" 'NO'
+opts_AddOptional '--mcsigma' 'CorrectionSigma' 'number' "myelin map bias correction sigma, default '$defaultSigma'" "$defaultSigma"
+opts_AddOptional '--matlab-run-mode' 'MatlabRunMode' '0, 1, or 2' "defaults to $g_matlab_default_mode
+0 = compiled MATLAB
+1 = interpreted MATLAB
+2 = Octave" "$g_matlab_default_mode"
+opts_ParseArguments "$@"
+
+if ((pipedirguessed))
+then
+    log_Err_Abort "HCPPIPEDIR is not set, you must first source your edited copy of Examples/Scripts/SetUpHCPPipeline.sh"
+fi
+
+# Verify required environment variables are set and log value
+log_Check_Env_Var HCPPIPEDIR
+log_Check_Env_Var CARET7DIR
+
+#display HCP Pipeline version
+log_Msg "Showing HCP Pipelines version"
+"${HCPPIPEDIR}"/show_version --short
+
+#display the parsed/default values
+opts_ShowValues
+
+# ------------------------------------------------------------------------------
+#  Main processing of script.
+# ------------------------------------------------------------------------------
+log_Msg "Starting main functionality"
+Caret7_Command=${CARET7DIR}/wb_command
+RegNameToUse=""
+if [[ "$RegName" != "" ]]; then
+	RegNameToUse=_${RegName}
+fi
+
+# default folders
+SubjFolder=${StudyFolder}/${Subject}
+AtlasSpaceFolder=${SubjFolder}/MNINonLinear
+T1wFolder=${SubjFolder}/T1w
+#NativeT1wFolder=${T1wFolder}/Native
+#NativeFolder=${AtlasSpaceFolder}/Native
+#DownSampleFolder=${AtlasSpaceFolder}/fsaverage_LR${LowResMesh}k
+#DownSampleT1wFolder=${T1wFolder}/fsaverage_LR${LowResMesh}k
+
+# convert upper case to lower case
+InputMesh=`echo "$InputMesh" | tr '[:upper:]' '[:lower:]'`
+OutputMesh=`echo "$OutputMesh" | tr '[:upper:]' '[:lower:]'`
+
+if [[ "$InputMesh" == "$OutputMesh" ]]; then
+	log_Err_Abort "the input mesh ${InputMesh} and the output mesh ${OutputMesh} shouldn't be the same!"
+fi
+
+case "$InputMesh" in
+	(native)
+		InputMeshString="native"
+		InputFolder=${AtlasSpaceFolder}/Native
+		InputT1wFolder=${T1wFolder}/Native
+		InputBCMapName="MyelinMap"
+		;;
+	(164)
+		InputMeshString=164k_fs_LR
+		InputFolder=${AtlasSpaceFolder}
+		InputT1wFolder=${AtlasSpaceFolder}
+		InputBCMapName="MyelinMap"
+		;;
+	(32)
```

Comment:
This doesn't support other meshes like 58k, or any new meshes we might make in the future.  The 32k case should probably use `(*)` as a catch-all, and use the argument assuming it is a number for `fsaverage_*k` and `*k_fs_LR`, etc.  You could use a regexp test to sanity check that it consists entirely of digits.

Reply:
Agree this should be agnostic to mesh resolution.

Reply:
Native is always in the native folder.  164k is always in the MNINonlinear folder.  Other meshes are in fsaverage_LR${Mesh} folder.